scripts/generate_abi.py

```diff
@@ -0,0 +1,97 @@
+import json
+import tempfile
+from pathlib import Path
+from typing import Any
+
+import click
+from psutil import Popen
+
+COMPILE_COMMAND = """
+git clone git@github.com:beamer-bridge/beamer.git {beamer_path} &&
+cd {beamer_path} &&
+git fetch &&
+git checkout {commit_id} &&
+$(which ape) compile
+"""
+
+
+def _get_contract_commit_ids(deployment_path: Path) -> dict[str, set[str]]:
+    file_contract_commits: dict[str, set[str]] = {}
+    for file_path in deployment_path.glob("*.json"):
+        with open(file_path, "r") as file:
+            data = json.load(file)
+            data.pop("deployer")
+            for contract_name, contract_info in data.items():
+                if contract_name not in file_contract_commits:
+                    file_contract_commits[contract_name] = set()
+                file_contract_commits[contract_name].add(contract_info["beamer_commit"])
+    return file_contract_commits
+
+
+def _compile_abis(commit_id: str, beamer_path: str) -> None:
+    Popen(
+        COMPILE_COMMAND.format(beamer_path=beamer_path, commit_id=commit_id).replace("\n", ""),
+        shell=True,
+    ).wait()
+
+
+def _get_contract_abis(
+    generated_abis: dict[str, dict[str, Any]]
+) -> dict[str, list[dict[str, Any]]]:
+    contract_abis: dict[str, list[dict[str, Any]]] = {}
+    for abis in generated_abis.values():
+        for contract_name, abi in abis.items():
+            if contract_name not in contract_abis:
+                contract_abis[contract_name] = []
+            contract_abis[contract_name].append(abi)
+    return contract_abis
```

Comment:
I have a feeling this function would not be needed if you modified the generated mapping in `_generate_abis` to look like
```
generated_abis[contract_name][commit_id]
```
instead of
```
generated_abis[commit_id][contract_name]
```

Put another way, this function should not be necessary if you generate the data in the desired shape directly in `_generate_abis`.

Reply:
i have done this just to use the if condition to pass some commits if i already processed them, so maybe i can just use another set and completely forget about commit_id here

Reply:
again i guess i need it, ok :D